### ISSUE TITLE
BsplineSE3.cpp timestamp_max Issue

### ISSUE THREAD

**Issue by zhangyanyu0722** [rpng/open_vins#280]
https://github.com/rpng/open_vins/blob/b12569c6697d1471da1523f1daec8c4f9e80f7fb/ov_core/src/sim/BsplineSE3.cpp#L54

Should it be ``` if (pose.first >= timestamp_max) { ``` ?

**Comment by goldbattle** [comment]
Yep! Nice catch! Old code probably still works as expected if the poses `trajectory_points` are sorted, but this should get fixed. Should probably also enforce the monotonic time / sorted poses also in that vector when reading from the file.

**Comment by goldbattle** [comment]
I will keep this open till I fix it.
